### DISCUSSION TITLE
All Redirect using navigate: true by default, And Events

### DISCUSSION THREAD

**Discussion by h-avatar** [livewire/livewire#6880] [7. Bugs (archived)] [answered]
### Livewire version

3.0.5

### Which PHP version are you using?

PHP 8.2

### Repository URL

https://github.com/h-avatar/livewire-check-navigate

### Steps To Reproduce

# Main Problem
Hi, I have some problems with livewire, First of all my all redirect with "$this->redirect()" do with navigate feature.
In sample repository, I created two route that one do redirect with navigate and another redirect without navigate, but both of them redirect with navigate.

First url:
`http://127.0.0.1:8000/first`
`$this->redirect('/second',navigate: true);`

Second url:
`http://127.0.0.1:8000/second`
`$this->redirect('/first');`

## Another Problem
When i dispatch event with "$this->dispatch" and then on js file need to change dom, I have to put my event on setTimeout(callback,1) with 1ms to work otherwise dom is not ready and i get error.

```js
Livewire.on('start_timer', () => {
    console.log($('.timer').length) // Output is 0

    setTimeout(function () {
        console.log($('.timer').length) // Output is 1
    },1);
});
```
### Unimportant problem
In the Navigate document, it is written that the "livewire:navigated" event is called both in the initial execution of the page and in the continuation. But for me, it is not called in the initial page load. I have to use both "livewire:navigated" event and "livewire:initialized" event at the same time.

![navigate](https://github.com/livewire/livewire/assets/29028910/a7e75377-8305-4ce7-8197-8c8c7e76cd58)

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by joshhanley** [comment, marked as answer]
@h-avatar thanks for reporting! There's already an open PR to fix this issue 🙂 see PR #6699
